opm/simulators/wells/RateConverter.hpp

```diff
@@ -488,14 +516,23 @@ namespace Opm {
                     auto& saltConcentration = ra.saltConcentration;
 
                     // sum p, rs, rv, and T.
-                    double hydrocarbonPV = pv_cell*hydrocarbon;
-                    if (hydrocarbonPV > 0) {
-                        pv += hydrocarbonPV;
-                        p += fs.pressure(FluidSystem::oilPhaseIdx).value()*hydrocarbonPV;
-                        rs += fs.Rs().value()*hydrocarbonPV;
-                        rv += fs.Rv().value()*hydrocarbonPV;
-                        T += fs.temperature(FluidSystem::oilPhaseIdx).value()*hydrocarbonPV;
-                        saltConcentration += fs.saltConcentration().value()*hydrocarbonPV;
+                    if (has_hc) {
+                        double hydrocarbonPV = pv_cell*hydrocarbon;
+                        if (hydrocarbonPV > 0) {
+                            pv += hydrocarbonPV;
+                            p += fs.pressure(phaseForProps).value()*hydrocarbonPV;
+                            rs += fs.Rs().value()*hydrocarbonPV;
+                            rv += fs.Rv().value()*hydrocarbonPV;
+                            T += fs.temperature(phaseForProps).value()*hydrocarbonPV;
+                            saltConcentration += fs.saltConcentration().value()*hydrocarbonPV;
+                        }
+                    } else {
+                        pv += pv_cell;
+                        p += fs.pressure(phaseForProps).value()*pv;
+                        rs += fs.Rs().value()*pv;
+                        rv += fs.Rv().value()*pv;
+                        T += fs.temperature(phaseForProps).value()*pv;
+                        saltConcentration += fs.saltConcentration().value()*pv;
```

Comment:
Did you really mean `*pv` here or was that supposed to be `*pv_cell` instead?

Reply:
This was a bug. Will ensure fixed when I refactor.